integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -9,238 +9,215 @@
+    heritage: {{ $release.Service }}
     role: alluxio-master
 spec:
   selector:
     matchLabels:
-      app: {{ template "alluxio.name" . }}
+      app: {{ $name }}
       role: alluxio-master
-      name: {{ template "alluxio.fullname" . }}-master
-  serviceName: "alluxio-master"
-  replicas: {{ .Values.master.replicas }}
+      name: {{ $fullName }}-{{ $masterName }}
+  serviceName: {{ $masterName | quote }}
+  replicas: {{ $root.Values.master.replicas }}
   template:
     metadata:
       labels:
-        name: {{ template "alluxio.fullname" . }}-master
-        app: {{ template "alluxio.name" . }}
-        chart: {{ template "alluxio.chart" . }}
-        release: {{ .Release.Name }}
-        heritage: {{ .Release.Service }}
+        name: {{ $fullName }}-{{ $masterName }}
+        app: {{ $name }}
+        chart: {{ $chart }}
+        release: {{ $release.Name }}
+        heritage: {{ $release.Service }}
         role: alluxio-master
     spec:
-      hostNetwork: true
-      dnsPolicy: ClusterFirstWithHostNet
+      hostNetwork: {{ $root.Values.master.hostNetwork }}
+      dnsPolicy: {{ $root.Values.master.dnsPolicy }}
       nodeSelector:
-      {{- if .Values.master.nodeSelector }}
-{{ toYaml .Values.master.nodeSelector | trim | indent 8  }}
-      {{- else if .Values.nodeSelector }}
-{{ toYaml .Values.nodeSelector | trim | indent 8  }}
+      {{- if $root.Values.master.nodeSelector }}
+{{ toYaml $root.Values.master.nodeSelector | trim | indent 8  }}
+      {{- else if $root.Values.nodeSelector }}
+{{ toYaml $root.Values.nodeSelector | trim | indent 8  }}
       {{- end }}
-      selector:
-        matchLabels:
-          app: {{ template "alluxio.name" . }}
-          role: alluxio-master
-          name: {{ template "alluxio.fullname" . }}-master
       securityContext:
-        runAsUser: {{ .Values.user }}
-        runAsGroup: {{ .Values.group }}
-        fsGroup: {{ .Values.fsGroup }}
+        fsGroup: {{ $root.Values.fsGroup }}
+      {{- if and $needJournalVolume (not $isRoot) }}
       initContainers:
-        - name: format-master-if-need
-          image: {{ .Values.image }}:{{ .Values.imageTag }}
-          imagePullPolicy: {{ .Values.imagePullPolicy }}
-          resources:
-            limits:
-            {{- if .Values.master.resources  }}
-              {{- if .Values.master.resources.limits }}
-                {{- if .Values.master.resources.limits.cpu  }}
-              cpu: {{ .Values.master.resources.limits.cpu }}
-              memory: {{ .Values.master.resources.limits.memory }}
-                {{- end }}
-              {{- end }}
-            requests:
-              {{- if .Values.master.resources.requests }}
-                {{- if .Values.master.resources.requests.cpu  }}
-              cpu: {{ .Values.master.resources.requests.cpu }}
-              memory: {{ .Values.master.resources.requests.memory }}
-                {{- end }}
-              {{- end }}
-            {{- end }}
```

Comment:
@cheyang I think `--no-format` option in Alluxio master container work as the journals are not formatted if they exist already. Please correct me if I'm wrong.

Reply:
@jiacheliu3 did you verify this is true for embedded as well?